pkg/controller/keyvault/kv_util.go

```diff
@@ -0,0 +1,116 @@
+package keyvault
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	kvcsi "github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/provider/types"
+	v1 "k8s.io/api/networking/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	secv1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+)
+
+var nginxNamePrefix = "keyvault-nginx-"
+
+func buildSPC(obj client.Object, spc *secv1.SecretProviderClass, config *config.Config) (bool, error) {
+	var certURI, specSecretName string
+
+	switch t := obj.(type) {
+	case *v1alpha1.NginxIngressController:
+		if t.Spec.DefaultSSLCertificate == nil || t.Spec.DefaultSSLCertificate.KeyVaultURI == nil {
+			return false, nil
+		}
+		certURI = *t.Spec.DefaultSSLCertificate.KeyVaultURI
+		specSecretName = DefaultNginxCertName(t)
+	case *v1.Ingress:
+		if t.Annotations == nil {
+			return false, nil
+		}
+
+		certURI = t.Annotations["kubernetes.azure.com/tls-cert-keyvault-uri"]
+		specSecretName = fmt.Sprintf("keyvault-%s", t.Name)
+	default:
+		return false, fmt.Errorf("incorrect object type: %s", reflect.TypeOf(obj).String())
```

Comment:
just print t here?

avoiding reflect when possible is good. it has some weird behaviors